python/paddle/distribution.py

```diff
@@ -102,21 +102,23 @@ def _to_tensor(self, *args):
         tmp = 0.
 
         for arg in args:
-            valid_arg = False
-            for cls in [float, list, np.ndarray, tensor.Variable]:
-                if isinstance(arg, cls):
-                    valid_arg = True
-                    break
-            assert valid_arg, "type of input args must be float, list, numpy.ndarray or Tensor."
             if isinstance(arg, float):
-                arg = np.zeros(1) + arg
+                arg = [arg]
+            elif not isinstance(arg, list) and not isinstance(arg, np.ndarray):
+                raise TypeError(
+                    "Type of input args must be float, list, numpy.ndarray or Tensor, but received type {}".
+                    format(type(arg)))
+
             arg_np = np.array(arg)
             arg_dtype = arg_np.dtype
             if str(arg_dtype) not in ['float32']:
```

Comment:
```suggestion
            if str(arg_dtype) != 'float32':
```

Reply:
And, the positive conditions is better than the negative conditions.

Reply:
Done